### PR TITLE
[doc] improve strong typing sentence

### PR DESCRIPTION
**Pull request by robstoll**
My go at improving the sentence about strong types which has an open TODO

### CODE REVIEW

**Comment by robstoll** [comment]
The build failure looks foreign to my change

**Comment by joroKr21** [comment]
> The build failure looks foreign to my change

The fix was merged: #1980

**Comment by robstoll** [comment]
@joroKr21 thanks, now I need a rebase-feature on github so that I can rebase my branch via web-gui :D

**Comment by robstoll** [comment]
please let me know in case a rebase of the branch is necessary. I would prefer if I don't have to checkout the project (do everything from the web-gui)

**Comment by joroKr21** [comment]
> please let me know in case a rebase of the branch is necessary.

I would assume so. I've also found myself often enough wishing GitHub would provide this UI.
I wonder if Codespaces would work in this project.

**Comment by SethTisue** [comment]
It's fine to merge this without worrying about the CI failure, since it's clearly unrelated.

However before we merge let's (hereby) ask @alvinj for a review, since this is an especially prominent part of the book.

Also there is one wording here I'm not really comfortable with: "where the compiler kind of writes code for you" — we don't require super-formal language on the site, but I think that's too loose a wording. perhaps "the compiler supplies method arguments for you"?

**Comment by robstoll** [comment]
@SethTisue I see your point but speaking only about method arguments is too narrowed IMO as the compiler does more e.g. in case of typeclass derivation. How about I change it to:
where the compiler synthesizes code for you (e.g. supplying method arguments in case of implicit parameters)

**Comment by martijnhoekstra** [comment]
Emphasizing what the programmer doesn't have to do, rather than what the compiler does, makes more sense to me for user-facing documentation.

In this case, that's omitting terms that can be inferred from their type and the context.

**Comment by robstoll** [comment]
@martijnhoekstra do you have a concrete idea how to phrase it?

**Comment by martijnhoekstra** [comment]
Maybe "Strong types allows you to omit things that the language can infer from the type and the context, like given parameters or the availability of extension methods"?

**Comment by martijnhoekstra** [comment]
`using` totally should have been called `context`. Let me quickly make a contributors thread for that.

**Comment by alvinj** [comment]
Hi @SethTisue, I’m a little swamped with a project deadline coming up this Tuesday, but I’ll be glad to look at this on Wednesday. Or, feel free to go with whatever you all think is good. 👍

**Comment by SethTisue** [comment]
no rush

File: _overviews/scala3-book/scala-features.md [hunk 1]
```diff
@@ -136,7 +136,7 @@ Because Scala is considered to be a [strong, statically-typed language](https://
 - Method type declarations tell readers what the method does, and help serve as documentation
 - Types make your code easier to maintain
 - Scalability: types help ensure correctness across arbitrarily large applications and development teams
-- Strong types enable Scala features like implicits (TODO: I need help on this wording and description)
+- Strong typing in combination with excellent inference enables mechanisms like [contextual abstraction]({{ site.scala3ref }}/contextual/motivation.html) that allows you to omit boilerplate code which can be inferred by the compiler based on type definitions and a given context (e.g. method arguments for implicit parameters).
```
Comment (b-studios) [inline _overviews/scala3-book/scala-features.md:139]:
```suggestion
- Strong typing in combination with excellent inference enables mechanisms like [contextual abstraction]({{ site.scala3ref }}/contextual/motivation.html) that allows you to omit boilerplate code. Often, this boilerplate code can be inferred by the compiler, based on type definitions and a given context (e.g. method arguments for implicit parameters).
```

**Review by b-studios — APPROVED** [review]
LGTM I would just break down this sentence, since it became quite long :)

File: _overviews/scala3-book/scala-features.md [hunk 2]
```diff
@@ -202,21 +202,23 @@ Scala is a functional programming (FP) language, meaning:
...
-Every variable is an object, and every “operator” is a method.
+Every variable is an object and every “operator” is a method.
 
 In Scala, all types inherit from a top-level class `Any`, whose immediate children are `AnyVal` (_value types_, such as `Int` and `Boolean`) and `AnyRef` (_reference types_, as in Java).
 This means that the Java distinction between primitive types and boxed types (e.g. `int` vs. `Integer`) isn’t present in Scala.
 Boxing and unboxing is completely transparent to the user.
 
 {% comment %}
-Add the “types hierarchy” image here?
+- AnyRef above is wrong in case of strict null checking, no? On the other hand, maybe too much information to state this here
```
Comment (b-studios) [inline _overviews/scala3-book/scala-features.md:219]: I agree it is too much info here.

File: _overviews/scala3-book/scala-features.md [hunk 3]
```diff
@@ -136,7 +136,7 @@ Because Scala is considered to be a [strong, statically-typed language](https://
 - Method type declarations tell readers what the method does, and help serve as documentation
 - Types make your code easier to maintain
 - Scalability: types help ensure correctness across arbitrarily large applications and development teams
-- Strong types enable Scala features like implicits (TODO: I need help on this wording and description)
+- Strong typing in combination with excellent inference enables mechanisms like [contextual abstraction]({{ site.scala3ref }}/contextual/motivation.html) that allows you to omit boilerplate code. Often, this boilerplate code can be inferred by the compiler, based on type definitions and a given context (e.g. method arguments for implicit parameters).
```
Comment (b-studios) [inline _overviews/scala3-book/scala-features.md:139]:
```suggestion
- Strong typing in combination with excellent inference enables mechanisms like [contextual abstraction]({% link _overviews/scala3-book/ca-contextual-abstractions-intro.md %}) that allows you to omit boilerplate code. Often, this boilerplate code can be inferred by the compiler, based on type definitions and a given context (e.g. method arguments for implicit parameters).
```

**Comment by b-studios** [comment]
This PR is lying around already quite some time and I apologize. Let me just merge it and let's open new PRs for further improvements.

**Pull request merged**